Hadrons/Modules/MRHQ/RHQInsertionI.hpp

```diff
@@ -109,17 +110,29 @@ void TRHQInsertionI<FImpl, GImpl>::setup(void)
 template <typename FImpl, typename GImpl>
 void TRHQInsertionI<FImpl, GImpl>::execute(void)
 {
-    LOG(Message) << "Applying Improvement term I to'" << par().q
+    LOG(Message) << "Applying Improvement term I with index " << par().index
+                 << " and gamma5=" << par().gamma5 
+                 << " to '" << par().q 
                  << std::endl;
 
-    auto &field    = envGet(PropagatorField, par().q);
-    const auto &gaugefield  = envGet(GaugeField, par().gauge);
-    const auto &gauge_index = par().gauge_index;
-
-    const auto internal_gauge = peekLorentz(gaugefield, gauge_index);
-    PropagatorField insertion = (GImpl::CovShiftForward(internal_gauge,gauge_index,field) - GImpl::CovShiftBackward(internal_gauge,gauge_index,field));
+    if (par().gamma5 != Gamma::Algebra::Gamma5 && par().gamma5 != Gamma::Algebra::Identity)
```

Comment:
Now I think about it, we should check with Felix whether adding if and switch statements will affect the GPU code - this is just a reminder for myself to do that before merging